miio/heater_miot.py

```diff
@@ -0,0 +1,209 @@
+import logging
+from typing import Any, Dict
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+_MAPPING = {
+    # Source https://miot-spec.org/miot-spec-v2/instance\?type\=urn:miot-spec-v2:device:heater:0000A01A:zhimi-mc2:1
+    # Heater (siid=2)
+    "power": {"siid": 2, "piid": 1},
+    "target_temperature": {"siid": 2, "piid": 5},
+    # Countdown (siid=3)
+    "countdown_time": {"siid": 3, "piid": 1},
+    # Environment (siid=4)
+    "temperature": {"siid": 4, "piid": 7},
+    # Physical Control Locked (siid=6)
+    "child_lock": {"siid": 5, "piid": 1},
+    # Alarm (siid=6)
+    "buzzer": {"siid": 6, "piid": 1},
+    # Indicator light (siid=7)
+    "led_brightness": {"siid": 7, "piid": 3},
+}
+
+
+class LedBrightness(enum.Enum):
+    On = 0
+    Off = 1
+
+
+class HeaterMiotException(DeviceException):
+    pass
+
+
+class HeaterMiotStatus:
+    """Container for status reports from the heater which uses the MIoT protocol."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is currently on."""
+        return self.data["power"]
+
+    @property
+    def target_temperature(self) -> int:
+        """Target temperature."""
+        return self.data["target_temperature"]
+
+    @property
+    def countdown_time(self) -> int:
+        """Countdown time."""
+        return self.data["countdown_time"]
+
+    @property
+    def temperature(self) -> float:
+        """Current temperature."""
+        return self.data["temperature"]
+
+    @property
+    def child_lock(self) -> bool:
+        """True if child lock is on, False otherwise."""
+        return self.data["child_lock"] is True
+
+    @property
+    def buzzer(self) -> bool:
+        """True if buzzer is turned on, False otherwise."""
+        return self.data["buzzer"] is True
+
+    @property
+    def brightness(self) -> LedBrightness:
+        """LED indicator brightness."""
+        return LedBrightness(self.data["led_brightness"])
+
+    def __repr__(self) -> str:
+        s = (
+            "<HeaterMiotStatus power=%s, "
+            "target_temperature=%s, "
+            "temperature=%s, "
+            "led_brightness=%s, "
+            "buzzer=%s, "
+            "child_lock=%s, "
+            "countdown_time=%s "
+            % (
+                self.power,
+                self.target_temperature,
+                self.temperature,
+                self.brightness,
+                self.buzzer,
+                self.child_lock,
+                self.countdown_time,
+            )
+        )
+        return s
+
+
+class HeaterMiot(MiotDevice):
+    """Main class representing the heater which uses MIoT protocol."""
```

Comment:
Please add the model also to this docstring to let readers to know which device(s) are supported.